pages/favs.jsx

```diff
@@ -3,18 +3,61 @@ import { getFavs } from '../js/favs-manager';
 import ProjectList from '../components/project-list/project-list.jsx';
 import HintMessage from '../components/hint-message/hint-message.jsx';
 
+import Service from '../js/service.js';
+import config from '../config/config.js';
+import env from "../config/env.generated.json";
+
 export default React.createClass({
   getInitialState() {
     return {
-      favs: null
+      favs: null,
+      loggedInCheckTimestamp: null
     };
   },
+  verifyLoggedInStatus() {
+    Service.nonce()
+      .then((response) => {
+        this.updateLoggedInInfo(true, JSON.parse(response).user);
```

Comment:
this JSON.parse should probably happen first, (either in a try/catch or with a comment that this can throw), with its data then passed into this.updateLoggedInInfo, for clarity.